api/src/main/java/jakarta/enterprise/lang/model/declarations/ParameterInfo.java

```diff
@@ -2,12 +2,40 @@
 
 import jakarta.enterprise.lang.model.types.Type;
 
+/**
+ * Provides read-only information about a method parameter.
+ */
 public interface ParameterInfo extends DeclarationInfo {
-    String name(); // TODO doesn't have to be known
+    /**
+     * Returns the zero-based position of this parameter in the method declaration.
+     *
+     * @return the position of this parameter in the method declaration
+     */
+    // TODO one-based?
+    int position();
 
+    /**
+     * Returns the name of this parameter, if it is known. Method parameter names may not always be known,
+     * in which case, an arbitrary string, or even {@code null}, may be returned.
+     *
+     * @return the name of this parameter, if it is known
+     */
+    // TODO add a method to determine whether the name is known? or just always return null when name is unknown?
+    String name();
```

Comment:
I would say it should be non null and if the arg name is not known `arg0`, `arg1` etc. should be returned

Reply:
Agree, done.